aws/billing.yml

```diff
@@ -0,0 +1,95 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: CloudFormation template to monitor AWS Free Tier usage, budget limits, and billing thresholds, with email notifications.
+
+Parameters:
+  budgetLimit:
+    Type: Number
+    Default: 20
+    Description: Monthly limit - will be notified when exceeding 75% and 100% of input value in USD
+  billingThreshold:
+    Type: Number
+    Default: 10
+    Description: Outstanding charges - will be notified when exceeding input value in USD
+  emailAddress:
+    Type: String
+    AllowedPattern: '[^@]+@[^@]+\.[^@]+'
+    Default: email@example.com
+  currency:
+    Type: String
+    Default: USD
+
+Resources:
+  notificationTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: BillingAndUsageNotifications
+      Subscription:
+        - Endpoint: !Ref emailAddress
+          Protocol: email
+  
+  freeTierAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: FreeTierExceeded
+      AlarmDescription: Triggers when AWS Free Tier limits are exceeded
+      MetricName: EstimatedCharges
+      Namespace: AWS/Billing
+      Statistic: Maximum
+      Period: 3600 # Check every hour
+      EvaluationPeriods: 1
+      Threshold: 0.01
+      ComparisonOperator: GreaterThanThreshold
+      AlarmActions:
+        - Ref: notificationTopic
+      Dimensions:
+        - Name: Currency
+          Value: !Ref currency
+
+  billingAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: BillingThresholdExceeded
+      AlarmDescription: Triggers when the AWS bill exceeds the specified threshold
+      MetricName: EstimatedCharges
+      Namespace: AWS/Billing
+      Statistic: Maximum
+      Period: 3600 # Check every hour
+      EvaluationPeriods: 1
+      Threshold: !Ref billingThreshold
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      AlarmActions:
+        - Ref: notificationTopic
+      Dimensions:
+        - Name: Currency
+          Value: !Ref currency
+
+  monthlyBudget:
+    Type: AWS::Budgets::Budget
+    Properties:
+      Budget:
+        BudgetName: MonthlyBudgetLimit
+        BudgetLimit:
+          Amount: !Ref budgetLimit
+          Unit: !Ref currency
+        TimeUnit: MONTHLY
+        BudgetType: COST
+        CostFilters: {}
+        CostTypes:
+          IncludeTax: true
+          IncludeSubscription: true
+          UseBlended: false
+      NotificationsWithSubscribers:
+        - Notification:
+            NotificationType: ACTUAL
+            ComparisonOperator: GREATER_THAN
+            Threshold: 75 # 75% of budget is used
+          Subscribers:
+            - SubscriptionType: EMAIL
+              Address: !Ref emailAddress
+        - Notification:
+            NotificationType: ACTUAL
+            ComparisonOperator: GREATER_THAN
+            Threshold: 100 # 100% of budget is used
+          Subscribers:
+            - SubscriptionType: EMAIL
+              Address: !Ref emailAddress
```

Comment:
This makes me curious if rate limiting is something we can easily add, just to cover our bases

Reply:
Will be moved to later PR